### ISSUE TITLE
camera controls: axis orientation and movement

### ISSUE THREAD

**Issue by mscipio** [maartenbreddels/ipyvolume#110]
Hi!
First of all, thanks for sharing this great tool!
Then, I have to say that I tried really hard going through the documentation and some of the issues here on github, but so far I have not been able to use the library much more than simply running a quickvolshow on my dataset ...

I have a couple of questions that I think should be quite easy to answer, but without knowing anything about JS, it's really hard for me to look through some of the source code. 
Here they are:

- I'd like to **set the orientation of the axes when showing the volume render** (i.e. like in standard cartesian, +x LR, +y front-back, +z bottom-up): is there some way I can tell quickvolshow() how to orientate the axes?

- second, I'd like to be able to **rotate the volume rendering around the (supposedly) vertical z-axis** (the dataset becomes too chaotic with a completely free orbit): how can I do it?

Thanks again, and I am sorry if these questions are duplicated somewhere else, but I wasn't able to find an answer for them ...

**Comment by maartenbreddels** [comment]
Dear Michele,

thanks for your kind words.
Do you mean you have trouble with the documentation as it is, or simply that you couldn't fine this issue you have in the documentation? If you have feedback on the documentation, I'm all ears, it kind of grew organically.

 * First point: not currently, but I have discussed this on some issue before. I'd like to support his, and making z go up should be the default. In any case it would be best to match Matlab/mayavi/matplotlib closely. It will be a 'breaking' change, but it needs to happen.

* Second point: now you can click the small icon in the toolbar (with the arrow up) to get that behaviour, you can also set it programmatically by changing `Figure.camera_control` (to 'orbit' instead of 'trackball').

Let me know if this helps!

**Comment by mscipio** [comment]
Hello! 
Let's see what I've understood

1. documentation seems ok, but I could use more examples about "custom" use (i.e. not using quick* functions) ... more on 3

2. Do you mean that there is no way I can customize axes orientation? Let's say I make my own fork, where can I find this information? or, second option, can I switch off axes labels (and do volume reorientation on my own)? 

3. I tried the toggle button but I saw no effect ... weird, I know! When you say I can set Figure.camera_control = 'orbit', what exactly do you mean? do I need to change the code for quickvolshow, or can I pass it as a parameter?

Thank for the help!!

**Comment by maartenbreddels** [comment]
No, not the orientation, but you can control axis labels and styling. The styling documentation isn't great/non-existing, it didn't really converge to something i'm totally happy with. 
Try: `ipv.style.use('nobox')` and see how the style is [defined](https://github.com/maartenbreddels/ipyvolume/blob/595caaf5be93e9b3d21e35a2e925c1f37b1fe179/ipyvolume/styles.py) here. `ipv.style.use([{'axes': {'visible':False}}])` will also work (btw, the extra list [] around it is actually a bug, shouldn't be needed)

If you want to work on this, I suggest adding a 4x4 (or maybe 3x3) matrix, [like these matrices](https://github.com/maartenbreddels/ipyvolume/blob/595caaf5be93e9b3d21e35a2e925c1f37b1fe179/ipyvolume/widgets.py#L139
), and name it something like 'basis_matrix'.  This would then need to be used here: https://github.com/maartenbreddels/ipyvolume/blob/595caaf5be93e9b3d21e35a2e925c1f37b1fe179/js/src/figure.js#L845 (my guess is before line 860).
Let me know if you want to proceed implementing this and if you need more help!

If you press the toggle button you can not go beyond the 'north pole', it should get 'stuck' there. And you can control it using sth like this:
```python
fig = ipv.figure()
fig.camera_control = 'orbit'
```

**Comment by mscipio** [comment]
Thanks for the hints, it will be easier now to look into the js code for what I need!

Just something I forgot to mention in the opening post: is it possible to **change the colormap** of the volume render? I know it may seem silly, but when dealing with medical images, having the possibility to adapt the colormap to data type / physician preference is kind of vital! ^^

**Comment by maartenbreddels** [comment]
Support is not great, however, look at this PR:
https://github.com/maartenbreddels/ipyvolume/pull/76
Also, on my TODO list is to be able to reproduce this:
http://graphicsrunner.blogspot.nl/2009/01/volume-rendering-102-transfer-functions.html?m=1

If you have a clear use case, like some data you'd like to visualize, and clear ideas on what kind of colormap to use, or how to generate it I'm all ears!

**Comment by mscipio** [comment]
Probably during the week I will find some time to seriously look at this.

I would say you to have a look to one of the notebook in this [repo](https://github.com/mscipio/occiput-suite/tree/master/occiput_suite/notebooks) of mine, but right now it seems GitHub is not able to load them ... I don't know if there is some problem with my connection or with the repository.

The idea was to integrate you volume rendering to show 3D/4D PET datasets (you would find mostly brain scans in the notebooks, but the anatomic region doesn't really matter). As for the colormap, usually in PET imaging you need gray, hot, viridis, ... but just having the option to choose one could be ok, I am not looking for something specific.

About how to generate them, that's probably the hard part, that way I was looking for the source code where the current one is selected (should be probably a rainbow or something, I guess ...)

If I have news, I will let you know.
Hope you'll do the same! ;)
